pystac_client/stac_api_io.py

```diff
@@ -263,7 +265,7 @@ def get_pages(
                 (link for link in page.get("links", []) if link["rel"] == "next"), None
             )
 
-    def assert_conforms_to(self, conformance_class: ConformanceClasses) -> None:
```

Comment:
I originally got rid of this `assert_conforms_to` because I was thinking that we would always want the user to be able to control whether or not to fail out if conformance fails. I think this very up for debate.